deployment/promotion/promote.py

```diff
@@ -9,396 +9,157 @@
-VERTA_CURL_OPTS: Options to pass to curl. Defaults to '-O'
+- VERTA_DEST_WORKSPACE: The workspace associated with the build on the destination Verta instance
+- VERTA_DEST_ORGANIZATION: The organization associated with the build on the destination Verta instance
+- VERTA_DEST_S3_BUCKET: The S3 bucket where the artifacts will be stored
+- VERTA_DEST_REGISTERED_MODEL_ID: [optional] The ID of the registered model to promote to. If missing, we'll create a new registered model
 
 With these values set, run the script. The script will not attempt to delete any data and will fail if the registered
 model (if an existing one has not been provided) or version already exists in the destination.
 """
 
 from locale import atoi
+from dataclasses import dataclass, field
+from typing import Optional
 
 import requests
 import os
 import datetime
 
-env_vars = ['VERTA_SOURCE_MODEL_VERSION_ID', 'VERTA_SOURCE_HOST', 'VERTA_SOURCE_EMAIL', 'VERTA_SOURCE_DEV_KEY',
-            'VERTA_SOURCE_WORKSPACE', 'VERTA_DEST_HOST', 'VERTA_DEST_EMAIL', 'VERTA_DEST_DEV_KEY',
-            'VERTA_DEST_WORKSPACE']
-opt_env_vars = ['VERTA_DEST_REGISTERED_MODEL']
-params = {}
-
-
-for param_name in env_vars:
-    param = os.environ.get(param_name)
-    if not param:
-        raise KeyError("Missing environment variable %s", param_name)
-    params[param_name] = param
-
-for param_name in opt_env_vars:
-    param = os.environ.get(param_name)
-    params[param_name] = param
-
-curl_opts = os.environ.get('VERTA_CURL_OPTS')
-if curl_opts:
-    params['VERTA_CURL_OPTS'] = curl_opts
+from verta import Client
+
+@dataclass
+class Config:
+    host: str
+    email: str
+    devkey: str
+    organization: Optional[str] = field(default=None)
+    workspace: Optional[str] = field(default=None)
+
+source_config = Config(
+    host=os.environ.get('VERTA_SOURCE_HOST'),
+    email=os.environ.get('VERTA_SOURCE_EMAIL'),
+    devkey=os.environ.get('VERTA_SOURCE_DEV_KEY'),
+    organization=os.environ.get('VERTA_SOURCE_ORGANIZATION'),
+    workspace=os.environ.get('VERTA_SOURCE_WORKSPACE')
+)
+
+dest_config = Config(
+    host=os.environ.get('VERTA_DEST_HOST'),
+    email=os.environ.get('VERTA_DEST_EMAIL'),
+    devkey=os.environ.get('VERTA_DEST_DEV_KEY'),
+    organization=os.environ.get('VERTA_DEST_ORGANIZATION'),
+    workspace=os.environ.get('VERTA_DEST_WORKSPACE')
+)
+
+def client_from_config(config):
+    client = Client(
+        host=config.host,
+        email=config.email,
+        dev_key=config.devkey,
+        organization_name=config.organization,
+    )
+
+    if config.workspace:
+        client.set_workspace(config.workspace)
+
+    return client
+
+print("Connecting to source Verta instance")
+source_client = client_from_config(source_config)
+print("Connecting to destination Verta instance")
+dest_client = client_from_config(dest_config)
+
+source_model_version_id = os.environ.get('VERTA_SOURCE_MODEL_VERSION_ID')
+if not source_model_version_id:
```

Comment:
fyi that it's generally better form to do `if source_model_version_id` is None`. In this case it doesn't matter, but if the value was `False` things would get weird.

Reply:
This is kind of an okay way to check for both `None` and `""` at the same time, which we'd probably want to handle in the same way

Reply:
We know for a fact that `source_model_version_id` will be a string or `None` here because of the line just above it. I'm ok with this idiom as it's pretty common 🤷